File: apps/problems/two-sum/boilerplate-full/function.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>
#include <climits>

##USER_CODE_HERE##

int main() {
  // The judge pipes the testcase in on stdin. It used to be read from a file
  // inside the Judge0 container, which is why this harness once needed the
  // problems directory bind-mounted into the sandbox.
  std::vector<std::string> lines;
  std::string line;
  while (std::getline(std::cin, line)) lines.push_back(line);

  int num1;
  std::istringstream(lines[0]) >> num1;
  int num2;
  std::istringstream(lines[1]) >> num2;
  int result = sum(num1, num2);
  std::cout << result << std::endl;
  return 0;
}

File: apps/problems/two-sum/boilerplate-full/function.rs
use std::io::{self, Read};
use std::str::Lines;

##USER_CODE_HERE##

fn main() -> io::Result<()> {
  // The judge pipes the testcase in on stdin.
  let mut input = String::new();
  io::stdin().read_to_string(&mut input)?;
  let mut lines = input.lines();
  let num1: i32 = lines.next().unwrap().parse().unwrap();
  let num2: i32 = lines.next().unwrap().parse().unwrap();
  let result = sum(num1, num2);
  println!("{}", result);
  Ok(())
}
